qmfkdldks/motion-menu-consumer
# Add product image carousel (consumer)

# What is the objective?

We help users to see what are the best options they can get from the shop.

# User story

As a user, I want to see what menu items the shop has so I decide what to eat.

# Acceptance criteria

- Users can swipe left/right to see more items.
- Lazy loading is applied to reduce initial load time.
- When the video product card is in view, it plays the video.
- It shows metadata, such as the total number of items.
- Product cards are clickable, and they will navigate to the product detail page.
- Add “More Item” which navigates to the plain text menu page.
- Check desktop view
- This is not updating the images page, it comes later

# Technical Notes

Some notes that can help developers and project managers. Or here you can explain how to implement the desired feature with a high-level description.

# What should be tested

- Scroll works fine
- The video product component is visible and playable
- Product cards are clickable and they are tracked
- Test Lazy loading feature

# QA Notes

Explain how to test newly implemented features and what are steps required.

1. Visit the shop review page [https://url.com/](https://url.com)
2. Check if reviews get rendered
3. Check sort and filter buttons

# Engineering Pitfalls
To add **media queries** from styles in line you can't do it this way:
```js
let carrousel_style: string = `
  @media (min-width: 768px) {
    width: ${wrapper_width}px;
  }
`;
```
You have to define css vars and pass it by style property: 
```js
 let carrousel_vars: string = `--wrapper-width: ${wrapper_width}px;`;
```
```js
 <div class="carrousel" style={carrousel_vars}>
 ```
 Then use it on styles:
```css
<style>
 @media (min-width: 768px) {
    .carrousel {
      position: relative;
      width: var(--wrapper-width);
    }
    }
</style>
```
